src/v/pandaproxy/rest/proxy.cc

```diff
@@ -22,51 +29,83 @@ namespace pandaproxy::rest {
 
 using server = ctx_server<proxy>;
 
+template<typename Handler>
+auto wrap(Handler h) {
+    return [h{std::move(h)}](
+             server::request_t rq,
+             server::reply_t rp) -> ss::future<server::reply_t> {
+        rq.authn_method = rq.service().config().authn_method(
+          rq.req->get_listener_idx());
+
+        if (rq.authn_method == config::rest_authn_method::http_basic) {
+            // Will throw 400 & 401 if auth fails
+            auto auth_result = rq.service().authenticator().authenticate(
+              *rq.req);
+            // Will throw 403 if user enabled HTTP Basic Auth but
+            // did not give the authorization header.
+            auth_result.require_authenticated();
```

Comment:
There are a few endpoints on the Admin API that do not require authentication but the authorization policy does. So `require_authenticated()` will throw 401 (not 403) with the message "Forbidden (authentication is required)".

nit: So, let's re-throw here with the expected 403 and a descriptive message.

Reply:
Addressed in https://github.com/redpanda-data/redpanda/pull/6626/commits/4033d97d6799083016ec45c87959d21b4f4194af